modules/nf-core/msisensor2/scan/main.nf

```diff
@@ -1,32 +1,31 @@
 process MSISENSOR2_SCAN {
-    tag "$fasta"
+    tag "${fasta}"
     label 'process_medium'
 
     conda "${moduleDir}/environment.yml"
-    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/msisensor2:0.1--hd03093a_0':
-        'biocontainers/msisensor2:0.1--hd03093a_0' }"
+    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
+        ? 'https://depot.galaxyproject.org/singularity/msisensor2:0.1--hd03093a_0'
+        : 'biocontainers/msisensor2:0.1--hd03093a_0'}"
 
     input:
-    path fasta
-    val output
+    tuple val(meta), path(fasta)
 
     output:
-    path output_path   , emit: scan
-    path "versions.yml", emit: versions
+    tuple val(meta), path("*.scan"), emit: scan
+    path "versions.yml",             emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args        = task.ext.args ?: ''
-    def inputs      = fasta.collect{ "-d $it"}.join(" ")
-    output_path = output ?: "output.scan"
+    def args   = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def inputs = fasta.collect { "-d ${it}" }.join(" ")
```

Comment:
Do we need a sort here? If more than one fasta is given they'll potentially arrive in different orders, which may affect output

Reply:
I don't think we need, but better be sage than sorry